rclcpp/include/rclcpp/service.hpp

```diff
@@ -137,7 +137,7 @@ class Service : public ServiceBase
     return std::shared_ptr<void>(new rmw_request_id_t);
   }
 
-  void handle_request(std::shared_ptr<void> & request_header, std::shared_ptr<void> & request)
+  void handle_request(std::shared_ptr<void> request_header, std::shared_ptr<void> request)
```

Comment:
`handle_request` is called internally by the executor which passes `void *` versions of `ServiceT::Request` and `rmw_request_id_t`. This could be changed to strongly type them and only recast them as `void *` when needed, though.